apps/rule-manager/client/src/ts/components/hooks/useRule.ts

```diff
@@ -67,6 +71,48 @@ export function useRule(ruleId: number | undefined) {
     setIsLoading(false);
   }
 
+  const publishRule = async (ruleId: number) => {
+    setIsPublishing(true);
+
+    try {
+      const response = await fetch(`${location}rules/${ruleId}/publish`, {
+        method: "POST",
+        headers: [["Content-Type", "application/json"]],
+        body: JSON.stringify({ reason: "Placeholder reason" })
+      });
+      if (!response.ok) {
+        throw new Error(`Failed to publish rules: ${response.status} ${response.statusText}`);
+      }
+
+      const rules: RuleDataFromServer = await response.json();
+      setRule({
+        draft: transformApiFormData(rules.draft),
+        live: rules.live.map(transformApiFormData)
+      });
+    } catch (error) {
+      setErrors(error);
+    } finally {
+      setIsPublishing(false);
+    }
+  }
+
+  const validateRule = async (ruleId: number) => {
+    setIsValidating(false);
+
+    try {
+      const response = await fetch(`${location}rules/${ruleId}/publish`);
+      if (response.status === 400) {
+        const validationErrors: Error[] = await response.json();
+        return setPublishingErrors(validationErrors);
```

Comment:
It looks like we're setting 'publishing errors' when validating, and setting just 'errors' when publishing. These states might be worth renaming for clarity.

Reply:
Renamed to `publishValidationErrors` in c0484b313f05a124793fb0e3cd8251f078b987c1.